meveo-admin/ejbs/src/main/java/org/meveo/service/admin/impl/MeveoModuleService.java

```diff
@@ -286,14 +286,52 @@ public void publishModule2MeveoInstance(MeveoModule module, MeveoInstance meveoI
         QueryBuilder qb = new QueryBuilder(MeveoModuleItem.class, "m");
         qb.addCriterion("itemCode", "=", code, true);
         qb.addCriterion("itemClass", "=", className, true);
-
         try {
             return (List<MeveoModuleItem>) qb.getQuery(getEntityManager()).getResultList();
         } catch (NoResultException e) {
             return null;
         }
     }
+    
+    @SuppressWarnings("unchecked")
+    public List<MeveoModuleItem> findModuleItem(String code, String className, String appliesTo) {
+    	List<MeveoModuleItem> res;
+    	QueryBuilder qb = new QueryBuilder(MeveoModuleItem.class, "m");
+    	qb.addCriterion("item_code", "=", code, true);
+    	qb.addCriterion("itemClass", "=", className, true);
+    	qb.addCriterion("appliesTo", "=", appliesTo, true);
+    	
+    	try {
+    		res = (List<MeveoModuleItem>) qb.getQuery(getEntityManager()).getResultList();
+    	} catch (NoResultException e) {
+    		res = null;
+    	}
+    	
+    	return res;
+    }
 
+    
+    /**
+     * Add module item with differentiate if appliesTo is null or not
+     * 
+     * @param meveoModuleItem Module item
+     */
+    public void addModuleItem(MeveoModuleItem meveoModuleItem) {
+    	List<MeveoModuleItem> testEmptyModule;
+    	MeveoModule module = meveoModuleItem.getMeveoModule();
+    	if (meveoModuleItem.getAppliesTo() == null) {
+    		testEmptyModule = this.findByCodeAndItemType(meveoModuleItem.getItemCode(), meveoModuleItem.getItemClass());
+    	}else {
+    		testEmptyModule = this.findModuleItem(meveoModuleItem.getItemCode(), meveoModuleItem.getItemClass(), meveoModuleItem.getAppliesTo());
+    	}
+    	if (testEmptyModule.isEmpty()) {
+    		module.getModuleItems().add(meveoModuleItem);
+    		meveoModuleItem.setMeveoModule(module);
+    	}else {
+    		throw new IllegalArgumentException("Impossible to add more than one ModuleItem");
```

Comment:
Message is not clear enough => "Module item with code XXX, (appliesTo XXX) already exists on module XXXX